scripts/Makefile.lib

```diff
@@ -296,7 +302,7 @@ cmd_dtc = echo '\#include "$(notdir $<)"' > dts/$(ARCH)/$(notdir $<)_pre_compile
 $(obj)/%.dtb: $(src)/%.dts FORCE
 	$(call if_changed_dep,dtc)
 
-$(obj)/%.dts_compiled: $(src)/%.dts FORCE
+$(obj)/%.dts_compiled: $(src)/%.dts $(DTS_OVERLAY) FORCE
```

Comment:
Yes.  The call to if_dep_changed will catch the dependency change if we also add in the overlay file.  Otherwise it won't ever run the cmd_dts.